### ISSUE TITLE
Popup gets unfullscreened when switching between gallery

### ISSUE THREAD

**Issue by qsniyg** [qsniyg/maxurl#611]
Thanks to JasonP on discord for reporting

In order to fix this, `makePopup` will probably have to contain a new mode for replacing an existing popup, as we can't re-request fullscreen outside of a user event (and due to the asynchronous nature of check_image_get, finding the next image is not considered part of the user event, regardless of how it was triggered)

**Comment by jasonpatel** [comment]
Anticipating the fix here, is this a challenging fix?

**Comment by qsniyg** [comment]
@jasonpatel Sorry for the wait, I'll try to work on it soon! It is indeed a bit challenging as it requires rewriting part of the way popups are handled (which is currently an extraordinarily messy part of the code). I have improved it somewhat through other changes, but it's still going to be a little complex to get right without breaking anything (though knowing this script, it'll _somehow_ break anyways, no matter the amount of testing I put in it haha).

This next release already has a number of possibly breaking changes (and is well overdue), so I'll aim for next release :)

**Comment by jasonpatel** [comment]
Cheers for reply, appreciate all the work your doing. This is a rather wonderful script :)

**Comment by 07416** [comment]
This doesn't only affect the userscript version? I've always experienced the same behavior in the Firefox addon (Linux, Fedora).

**Comment by qsniyg** [comment]
@07416 Yes. `topic:userscript` is the lowest common denominator (the extension uses the userscript at its core, but augments it with extra functionality). `topic:extension` is used for issues that only apply to the extension version.
